src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs

```diff
@@ -769,9 +769,7 @@ protected virtual void Dispose()
 
         public async Task<long> CalculateBackupSizeInBytes()
         {
-            var zipPath = GetBackupLocation();
-             zipPath = Path.Combine(zipPath, RestoreFromConfiguration.LastFileNameToRestore);
-            using (var zip = await GetZipArchiveForSnapshotCalc(zipPath))
+            using (var zip = await GetZipArchiveForSnapshotCalc(RestoreFromConfiguration.LastFileNameToRestore))
```

Comment:
Where is the NRE fix?

Reply:
Path.Combine was messing up the paths if we restore destination is cloud storage, if the data not found on cloud storage we return null